domain/src/main/scala/BitcoinCryptography.scala

```diff
@@ -1,13 +1,13 @@
 package org.parsec.protocol
 
-trait BitcoinCryptography {
-  // TODO: implement me
-  def bitcoinP2PKHAddressFromPublicKey(key: String): String = {
-    "btc-public-key"
-  }
+import org.apache.commons.codec.binary.Hex
+import org.bitcoinj.core.{ECKey, NetworkParameters}
 
-  // TODO: implement me
-  def bitcoinP2SHAddressFromPublicKey(key: String): String = {
-    "btc-public-key"
+trait BitcoinCryptography {
+  def bitcoinAddressFromPublicKey(key: String): String = {
```

Comment:
Nice. We'd be able to add `address` as a method on the base sealed trait `PublicKey` in that case 😉